### DISCUSSION TITLE
vega-view

### DISCUSSION THREAD

**Discussion by arnolddevos** [nushell/nushell#13620] [Show and tell]
Hi, here is a simple visualisation tool called [vega-view](https://github.com/cuprous-au/vega-view) that I created. It displays [nushell](https://www.nushell.sh/) tables using [Vega Lite](https://vega.github.io/vega-lite/) in a webview.  vega-view operates from the command line and pops open a window for the visualisation.  It does not rely on the network or web browser.  

For example, if you have a source of data such as a file `example.json` the command `open example.json | vega view (vega bar b --category=t --subcategory=a)` pops up a window with a bar graph.  

The advantage of combining nushell with vega is that nushell can source and manipulate many forms of data which vega-lite can visualise in many different ways.

**Comment by fdncred** [comment]
That's cool! Thanks for sharing.

**Comment by kubouch** [comment]
Hi, it looks really cool. Two quick notes:
1. In the `export-env` in `vega.nu`, it doesn't take into account if you have CARGO_TARGET_DIR set, I had to edit it manually.
2. When I run the bar graph example, it opens a new window but nothing shows in it. Running on Ubuntu 22.04. I installed the prerequisites listed in the wry project.

**Reply by arnolddevos** [reply]
Hi,

1. Right.  I can fix that.   A broader question is: what is the right way to distribute some nushell modules as part of a cargo package?
2. I will attempt to reproduce.

**Reply by ghost** [reply]
> what is the right way to distribute some nushell modules as part of a cargo package?

no package manager for modules has reached maturity (and not yet been deprecated again)

package managers, which are being developed:

* [nupm](https://github.com/nushell/nupm) (by the nushell team)
* [numng](https://github.com/jan9103/numng) (by me)

since both support the [nupm package format](https://github.com/nushell/nupm/blob/main/docs/design/references/METADATA.md) that's probably a safe bet

**Comment by NotTheDr01ds** [comment]
This looks amazing.  In the last week a new [Showcase channel](https://discord.com/channels/601130461678272522/1277584399323627572) has been added to the Nushell Discord server for showcasing stuff like this - I'd encourage you to post it there as well!
